Library/SpecUI/CommonUI/CWin/cwinFieldInit.asm

```diff
@@ -512,6 +515,30 @@ OLFieldEnsureExpressMenu	proc	far
 	mov	di, ds:[si]
 	add	di, ds:[di].Vis_offset
 	mov	ds:[di].OLFI_expressMenu, dx	; store chunk handle of menu
+
+if _MOTIF and TOOL_AREA_IS_TASK_BAR
+	;
+	; upscale E-Menu Button width a little in Motif / Taskbar, for aesthetic pleasure
+	; if TaskBar == on
+	; RUNTIME
+	;
+	push	ds					; save ds
+	segmov	ds, dgroup				; load dgroup
+	test	ds:[taskBarPrefs], mask TBF_ENABLED	; test if TBF_ENABLED is set
+	pop	ds					; restore ds
+	jz	done					; skip if no taskbar
+
+	push	si
+	mov	si, ds:[di].OLFI_expressMenu		; set *ds:si = menu
+	mov	ax, HINT_FIXED_SIZE			; ax - Vardata type
+	mov	cx, size SpecWidth + size SpecHeight + size word ; cx - Size of extra data. (cx = zero if no extra data).
+	call	ObjVarAddData				; returns: pointer to extra data in ds:bx
+	mov	ds:[bx].SSA_width, SpecWidth <SST_PIXELS, 30>

```

Comment:
Is giving a fixed size a good choice here? Think about different screen resolutions and express menu/taskbar sizes....

Reply:
Yup, you're right, but the question is, what would be a good alternative?

Reply:
Okay, I've reworked the code to make the width of the E-Menu Button dependent on the field size (width). However, after many tries, I think the best solution for the height is to stay at 20 pixels as this creates the best look. I've tested several resolutions and it works pretty in-line with the rest of Motif.